### DISCUSSION TITLE
ImageMagick Windows XP Portable version available for download

### DISCUSSION THREAD

**Discussion by Geo-Van** [ImageMagick/ImageMagick#7517] [General]
I try many hours to find and download the ImageMagick **_Portable version for Windows XP_**.
Note that according to my findings, the _latest working version for XP is the version 6.7.xx(TESTED and is working) and may be 6.8.xx_

I found only the following links which are not OFFICIAL download links:
https://www.imagemagick.org/discourse-server/viewtopic.php?t=29301
https://web.archive.org/web/20110921004715/http://mhonarc.veidrodis.com/image_magick/binaries/

Note that the OFFICIAL download for version 6 it has only the version 6.9.13-14  -- which is not working on Windows XP:
https://legacy.imagemagick.org/script/download.php

I know that Windows XP is not supported anymore, but because some people working with them,  I will kindly request to ask from IMAGEMAGIK to make at least one OFFICIAL download link of   ImageMagick **_Portable version for Windows XP_** - versions 6.7.xx or 6.8.xx.
I will appreciate it very much  if you know any official and trusted places to download ImageMagick **_Portable version for Windows XP_** - versions 6.7.xx or 6.8.xx.

**Comment by dlemstra** [comment]
We are not longer supporting Windows XP and we will not be going to anymore. We have stopped doing this a couple years ago.

**Comment by Geo-Van** [comment]
Thank you for your reply, i understand, but i will kindly request if someone knows a trusted place to download at least ONLY 1 old version of WINDOWS PORTABLE 6.7xx or 6.8xx, please

**Comment by Geo-Van** [comment]
Please further note that according to this:
https://github.com/ImageMagick/ImageMagick/discussions/2376

The latest working versions are : (I try them and yes are working under XP):
6.9.11-19-Q8-x86
7.0.10-19-portable-Q16-x86
which as stated there can be downloaded from:
http://martinleese.epizy.com/MyTemporaryDownloads/#WINDOWSXP

But, please, is there any OFFICIAL TRUSTED download link for only ---> **_7.0.10-19-portable-Q16-x86_**    ?
Please, if you can officially provide here a download link, we will appreciate it very much, please.

**Reply by dlemstra** [reply]
We don't have an archive of our old downloads so we cannot help you with that.

**Reply by Geo-Van** [reply]
Hello, 
Many thanks, i understand and thank you for response to my warnings. 

Please let me ask:
Since the ORIGINAL SOURCE CODE of 7.0.10-19 is OFFICIALLY available at :
https://github.com/ImageMagick/ImageMagick/releases?page=15

Is there any way to pay in order to "PRODUCE" from that SOURCE CODE the desired **_7.0.10-19-portable-Q16-x86_**    ?
I love the software, and i will be very glad if the last working XP version will be officially available - even here in this post.

I am looking forward for your response, please.

**Reply by dlemstra** [reply]
It might be possible to create the binaries with the help of Visual Studio and the latest source code. But you will need to build this with Visual Studio 2017 and I suspect you don't have access to this. And also I suspect that some projects will not build when it is build with Windows XP support. This is probably going to take quite a lot of time and the person that is going to do this will probably going to ask something in the $200 range and I suspect that is a lot more than you are expecting to pay for this.

**Reply by Geo-Van** [reply]
Thanks for your reply.
Indeed the price is too high, but please let me further ask:

**Question 1:**
Creating by myself the desired 7.0.10-19-portable-Q16-x86 from SOURCE CODE using Visual Studio:
https://imagemagick.org/script/install-source.php#windows
There are 2 free  Visual Studio editions, the  Visual Studio Community Edition 2017 and the  Visual Studio Code 2017.
Which one i must use ?
Or in order to do the job i must purchase the PAID  Visual Studio 2017 ?


**Question 2:**
I also have an idea to grab the "magick.exe" from the 7.0.10-19-portable-Q16-x86 , by grab it from third party softwares that uses the ImageMagick as back-end. Do you know any third party softwares that uses the ImageMagick, please?

Appreciate very much your help! 
Thank you!

**Reply by dlemstra** [reply]
I am not sure if you noticed this:

>Older versions require an active Visual Studio Subscription.

And that is a lot more than the other amount.

I have no clue what you meant with your second question.

**Reply by Geo-Van** [reply]
(1) I have already install in my PC machine Visual Studio Community Edition 2017 and the Visual Studio Code 2017 which are FREE, no subscription is required, but my question is that --> Can these two Visual Studio editions do the job, or do i must PURCHASE a Visual Studio subscription, please?

(2) Do you know any third party softwares that they use your command line imagemagick software to their GUI, please?

**Reply by dlemstra** [reply]
Well I guess you are lucky then, the community edition of Visual Studio 2017 can no longer be downloaded. You should be able to build ImageMagick with a community edition but how you need to do that is something you will need to figure out yourself.

There are probably companies that are using our free software but I don't know which application uses one that still works on Windows XP.

**Reply by Geo-Van** [reply]
(1) Thank you very much!

(2) Does not matter if works on XP, i will be glad to know some applications that uses your software. Please let me know if you know such applications names. Thank you.

**Comment by snibgo** [comment]
> Is there any way to pay in order to "PRODUCE" from that SOURCE CODE the desired 7.0.10-19-portable-Q16-x86 ?

https://github.com/ImageMagick/ImageMagick/archive/refs/tags/7.0.10-19.zip can be downloaded and built with Cygwin tools (./configure, make, make install), using current delegate libraries.

```
Version: ImageMagick 7.0.10-19 Q32 x86_64 2024-07-28 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(4.5)
Delegates (built-in): bzlib cairo fftw fontconfig fpx freetype heic jbig jng jpeg lcms ltdl lzma pangocairo png raqm raw rsvg tiff webp wmf x xml zlib
```
The result runs under Windows 11. I don't know if it would run under Windows XP. I don't distribute binaries.

**Reply by Geo-Van** [reply]
Many many thanks for your answer.
I try many hours, but because i am not a programmer, i did not find a way how to do it, but i will check for some online tutorials how to use Cygwin tools. (If someone knows of a good tutorial we will appreciate link it here).

Please let me ask regarding the SOURCE CODE:
As i see there are TWO DIFFERENT SOURCE CODES as follows:

**SOURCE CODE 1:**
https://legacy.imagemagick.org/script/install-source.php#:~:text=Install%20from%20Windows%20Source,-Building%20ImageMagick%20source&text=Next%2C%20launch%20your%20Visual%20Studio,on%20completion%20run%20the%20program.
AND from:
https://imagemagick.org/archive//windows/
WE download the SOURCE CODE that it is about 400MB


**SOURCE CODE 2:**
SOURCE CODE of 7.0.10-19 is OFFICIALLY available at :
https://github.com/ImageMagick/ImageMagick/releases?page=15
WE download the SOURCE CODE that it is about ONLY  16MB

**Here, please is my questions regarding the above TWO SOURCE CODES:**
(1) There is a very big difference in their size --> 400MB to 16MB  --> Why?

(2) SOURCE CODE 1 can only be used with VISUAL STUDIO 2017?
     There is no way to use it (in order to create windows binaries) with VISUAL STUDIO 2015 or older?
     There is no way to use it (in order to create windows binaries) with Cygwin tools?

(3) SOURCE CODE 2 can only be used (in order to create windows binaries) with Cygwin tools? 
     There is no way to use it (in order to create windows binaries) with Visual Studio 2017, please?

Thanks very much appreciate your help,
Best Regards

**Comment by snibgo** [comment]
The big source code contains ImageMagick plus delegate libraries to read and write PNG and JPG etc. The small source code does not contain the delegates.

A tutorial is: [Compiling IM with Cygwin](http://im.snibgo.com/compim.htm).

I know nothing about Visual Studio.

**Reply by Geo-Van** [reply]
Thank you very much!
I try the Cygwin method, as follow (according to the wonderful tutorial you link) - BIG THANK YOU for that:

**_Steps:_**   (Note - substitute User with your User name under the following commands ---> C:\cygwin64\home\User\ )

1. Install Cygwin with ALL the  "Devel" (development) package,

2. Run the following commands in Cygwin command window: (to create a folder "winim" at the C:\cygwin64\home\User\ )
    mkdir winim

3. I download the https://github.com/ImageMagick/ImageMagick/archive/refs/tags/7.0.10-19.zip , i unzip it and i copy-paste the unziped folder of "ImageMagick-7.0.10-19" at the DIRECTORY ------> C:\cygwin64\home\User\  

So, until now we have the following:
C:\cygwin64\home\User\winim
and
C:\cygwin64\home\User\ImageMagick-7.0.10-19

4. Now, back to Cygwin command window, i run the following 4 commands:

cd "ImageMagick-7.0.10-19"

   ./configure --prefix=/home/User/winim --with-quantum-depth=16 --enable-hdri --with-jp2=yes --with-rsvg=yes --with-windows-font-dir=/cygdrive/c/Windows/Fonts

   make

   make install


The above should create the desired creation under ---> C:\cygwin64\home\User\winim\

BUT, unfortunately no success, there are the following errors occur:
(a) During "make" command -----> Error occurs as follow:
_**make[1]: *** [Makefile:9891: coders/MagickCore_libMagickCore_7_Q16HDRI_la-jpeg.lo] Error 1
make[1]: Leaving directory '/home/User/ImageMagick-7.0.10-19'
make: *** [Makefile:5988: all] Error 2**_

(b) During "make install" command -----> Error occurs as follow:
**_make[1]: *** [Makefile:9891: coders/MagickCore_libMagickCore_7_Q16HDRI_la-jpeg.lo] Error 1
make[1]: Leaving directory '/home/User/ImageMagick-7.0.10-19'
make: *** [Makefile:12266: install] Error 2_**

And NOTHING is created under -->  C:\cygwin64\home\User\winim\
I do not know where i do a mistake.....
Can you please show me where is my mistake, please?

**Comment by snibgo** [comment]
If `make` is required, then `make install` will run it. If `make` is required and run but fails, then `make install` will fail with the same error. This has happened for you.

You didn't run `automake` and `autoconf` (in bash)? These are probably needed. Run these, then your `./configure`, `make` and `make install`. If any of these 5 commands fail, it probably needs to be fixed before the next one will succeed.

**Reply by Geo-Van** [reply]
Hello,
 I unzip  ImageMagick-7.0.10-19.zip  inside ---> C:\cygwin64\home\User\
 and i run the following  COMMANDS:

mkdir winim

cd ImageMagick-7.0.10-19

automake

autoconf  

./configure --prefix=/home/User/winim --with-quantum-depth=16 --enable-hdri --with-jp2=yes --with-rsvg=yes --with-windows-font-dir=/cygdrive/c/Windows/Fonts


make


BUT error occurs HERE as follow:
$ make
make: *** No targets specified and no makefile found.  Stop.


Please i will appreciate your help to overcome this error.Many thanks

**Comment by snibgo** [comment]
Does the large text file `config.log` look okay?

> ... and no makefile found.

That suggests automake failed, or you are in the wrong directory.

In the directory you ran `make`, try the Windows command `dir /od makefile*`.

Expect a output like this:
```
18/07/2024  20:20            11,088 Makefile.am
21/07/2024  12:16         1,459,702 Makefile.in
21/07/2024  12:18         1,187,074 Makefile
```

**Reply by Geo-Van** [reply]
Thanks a lot trying to help me. 
Unfortunately only TWO files are there  Makefile.am and  Makefile.in
Anyway, i spend a lot of time on this..... 
It is very dissapointing that IMAGEMAGIC has not at least the LATEST working version on XP available for download.
It will be nice if there is.
Once again many many thanks for your help!
